internal/groupsuffix/groupsuffix.go

```diff
@@ -61,19 +65,61 @@ func New(apiGroupSuffix string) kubeclient.Middleware {
 			rt.MutateRequest(mutateOwnerRefs(Replace, apiGroupSuffix))
 		}),
 
+		kubeclient.MiddlewareFunc(func(_ context.Context, rt kubeclient.RoundTrip) {
+			// we only care if this is a create on a TokenCredentialRequest without a subresource
+			if rt.Resource() != loginv1alpha1.SchemeGroupVersion.WithResource("tokencredentialrequests") ||
+				rt.Verb() != kubeclient.VerbCreate ||
+				rt.Subresource() != "" {
+				return
+			}
+
+			// we only do this on the way out, since on the way back in we don't set a spec in our
+			// TokenCredentialRequest
+			rt.MutateRequest(func(obj kubeclient.Object) error {
+				// if this cast fails, then i believe we have a so-called "programmer error"
+				// also this would be caught by the integration tests
+				tokenCredentialRequest := obj.(*loginv1alpha1.TokenCredentialRequest)
```

Comment:
Error, don't `panic`.  Our integration tests will never cover 100% of runtime cases.

Reply:
Agreed that we shouldn't panic. Perhaps we should remove some other `panic`'s from our code...for another day. 0c65284aec7ff90fee68b2e039f1f85abf6491bd.